src/common/meta/src/rpc/router.rs

```diff
@@ -266,6 +267,13 @@ impl TableRoute {
             })
             .collect()
     }
+
+    pub fn find_region_leader(&self, region_number: RegionNumber) -> Option<Peer> {
```

Comment:
This function may be inefficient when route info is large, I think we can build a map that keeps `<region_number, Peer>` when route info is building.